connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -281,9 +281,18 @@ private void readToLogEnd() {
             Iterator<Map.Entry<TopicPartition, Long>> it = endOffsets.entrySet().iterator();
             while (it.hasNext()) {
                 Map.Entry<TopicPartition, Long> entry = it.next();
-                if (consumer.position(entry.getKey()) >= entry.getValue())
+                TopicPartition topicPartition = entry.getKey();
+                Long endOffset = entry.getValue();
```

Comment:
unboxing will happen in the comparison in the `if` branch anyways, so probably better to do it early declaring the type `long` here.

Reply:
Fair enough; added.